src/lib/dd_profiling.cc

```diff
@@ -131,11 +134,13 @@ static int exec_ddprof(pid_t target_pid, pid_t parent_pid, int sock_fd) {
 
   kill(parent_pid, SIGTERM);
 
-  if (fexecve(fd, argv, environ) == -1) {
-    return -1;
+  if (const char *ddprof_exe = getenv(k_profiler_ddprof_exe_env_variable);
+      ddprof_exe) {
+    execve(ddprof_exe, argv, environ);
```

Comment:
Do we still want to do the write and create an fd operation ?

Reply:
I will move the memfd inside the if branch.